src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -1,6 +1,12 @@
 import React, { useEffect, useState } from 'react'
 import { Dropdown, Label } from 'semantic-ui-react'
+import { ResponseFull } from '../../../utils/types'
 import { ApplicationViewProps } from '../../types'
+import { useLanguageProvider } from '../../../contexts/Localisation'
+
+type ObjectOption = { [key: string]: any }
+type ObjectOptions = { options: ObjectOption[]; optionsDisplayProperty: string }
+type StringOptions = { options: string[]; optionsDisplayProperty: never }
```

Comment:
The point of these types is that it allows us to tell typescript that an "optionsDisplayProperty" parameter should always be paired with an options list made of objects, and vice-versa.